File: challenge_2/python/mjuiuc/SingleNumber.py
from collections import defaultdict
class Solution(object):
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        
        #Lazy solution here... exceeds runtime of course
        # for i in nums:
        #     if nums.count(i) == 1:
        #         return i
        
        #Do this without using extra memory? That's too hard. K well this worked and beat 50% cases so :/
        #Basically took a histogram of the list and then go back through it to find the number whose count
        #was one
        histo = defaultdict(int)
        for num in nums:
            histo[num] += 1
        for num in nums:
            if histo[num] == 1:
                return num

File: challenge_2/ruby/selectivealso/single_number.rb
nums = [2,3,4,2,3,5,4,6,4,6,9,10,9,8,7,8,10,7]

class Array
  def find_unique
    count = Hash.new(0)

    self.each do |num|
      count[num] += 1
    end

    return count.key(1)

  end
end


puts nums.find_unique
